test/dynamo/test_opinfo_dynamic.py

```diff
@@ -0,0 +1,208 @@
+# Owner(s): ["module: dynamo"]
+import atexit
+import torch
+import functools
+from functools import partial
+import unittest
+from torch.testing._internal.common_device_type import (
+    instantiate_device_type_tests,
+    onlyNativeDeviceTypes,
+    onlyCPU,
+    onlyCUDA,
+    OpDTypes,
+    ops,
+    skipCPUIf,
+    skipCUDAIf,
+)
+from torch.testing._internal.common_methods_invocations import op_db, skipOps
+from torch.testing._internal.common_utils import (
+    dtype_abbrs,
+    IS_MACOS,
+    IS_X86,
+    skipCUDAMemoryLeakCheckIf,
+    skipIfCrossRef,
+    skipIfTorchDynamo,
+    suppress_warnings,
+    TEST_WITH_ASAN,
+    TEST_WITH_ROCM,
+    TestCase,
+)
+from torch.testing._internal.inductor_utils import HAS_CPU, HAS_CUDA
+from unittest.mock import patch
+from torch._dynamo.test_case import run_tests
+
+
+
+
+# START = 0
+# END= 2000 
+
+
+records = []
+specialized_records = []
+def print_records():
+    from tabulate import tabulate
+    import csv
+    global records, specialized_records
+    records = set(records)
+    specialized_records = set(specialized_records)
+    print(tabulate(records))
+    print(tabulate(specialized_records))
+    with open("status.csv", "w") as f:
+        writer = csv.writer(f)
+        writer.writerow(("op", "status", "exception"))
+        writer.writerows(records)
+    with open("specialized.csv", "w") as f:
+        writer = csv.writer(f)
+        writer.writerow(("op", "specialized_input"))
+        writer.writerows(specialized_records)
+
+
+
+atexit.register(print_records)
+def clone_preserve_strides(x, device=None):
+    if not isinstance(x, torch.Tensor):
+        return x
+    buffer = torch.as_strided(
+        x, (x.untyped_storage().size() // x.element_size(),), (1,), 0
+    )
+    if not device:
+        buffer = buffer.clone()
+    else:
+        buffer = buffer.to(device, copy=True)
+    out = torch.as_strided(buffer, x.size(), x.stride(), x.storage_offset())
+    return out
+
+
+def check_model(
+    self,
+    op,
+    example_inputs,
+    kwargs=None,
+    *,
+    atol=1e-5,
+    rtol=1e-5,
+):
+
+    func = op.get_op()
+
+    def fn(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    model = fn
+
+    kwargs = kwargs or {}
+    torch._dynamo.reset()
+
+    ref_inputs = [clone_preserve_strides(x) for x in example_inputs]
+    ref_kwargs = kwargs
+    torch.manual_seed(0)
+
+    correct = model(*ref_inputs, **ref_kwargs)
+    called = False
+
+    def detect_specialization(gm, *args):
+        for node in gm.graph.nodes:
+            example_value = node.meta.get("example_value", None)
+            if node.op == "placeholder" and isinstance(example_value, torch._subclasses.FakeTensor):
+                size = example_value.size()
+                for sz in size:
+                    if isinstance(sz, int) and sz not in (0, 1):
```

Comment:
This is insufficient. You may have a SymInt which actually has been specialized into an int. In fact, I would expect this to miss most specializations